### DISCUSSION TITLE
Documentation: correct way to get client IP?

### DISCUSSION THREAD

**Discussion by sabard** [litestar-org/litestar#1319] [Q&A] [answered]
FastAPI has a [simple mechanism for accessing client information](https://stackoverflow.com/a/66015976) from the `Request` object. It looks like this information is stored in `ASGIConnection` in Starlite, but it's not clear to me how to access this object from a HTTP handler from the documentation, only middleware.

If this is easily done can an example be added to the documentation? Let me know if I'm missing something simple.

**Comment by Goldziher** [comment, marked as answer]
This is not FastAPI specific. This info is part of the ASGI spec and is as such accessible as part of the ASGI scope dictionary. 

You can access `request.client` to access it. Simply have the request instance injected into the route handler (or the websocket instance) to interact with it.

**Comment by sabard** [comment]
Got it. I see now that `Request` inherits from `ASGIConnection`. Thanks!
